src/srun/libsrun/fname.c

```diff
@@ -217,6 +218,19 @@ extern fname_t *fname_create(srun_job_t *job, char *format, int task_count)
 				break;
 			case 'x':
 				xmemcat(name, q, p - 1);
+                jobname = getenvp(job->env, "SLURM_JOB_NAME");
+                if (jobname != NULL) {
+                    jobnamecopy = strdup(jobname);
```

Comment:
Preferably, you define the copy `jobnamecopy` here -- if allowed by the (recent) compiler -- and limit its scope to this if-clause. No need to set it to NULL then after the free.

Reply:
Also, use `xstrdup`.